hypothesis-python/tests/py3/test_lookup.py

```diff
@@ -599,3 +599,48 @@ def test_bytestring_is_valid_sequence_of_int_and_parent_classes(type_):
         st.from_type(typing.Sequence[type_]),
         lambda val: isinstance(val, typing.ByteString),
     )
+
+
+def supports_protocol(protocol, inst):
+    # in python < 3.8 Protocols cannot be used with isinstance
+    # this check should work for all versions.
+    return issubclass(type(inst), protocol)
+
+
+def supports_casting(typ, thing):
```

Comment:
```python
from hypothesis.strategies._internal.types import can_cast
```

since you already wrote the function :grin:

Reply:
oh yeah good one

Reply:
> > I've just remembered `SupportsAbs` and `SupportsRound` need to be able to accept return type arg...
> 
> Good catch! This was broken before too... if you want to finish this one up tonight I'd be perfectly happy to merge it and then do another (patch) release with this fix later.

Yeah sorry haven't had a chance to work on it.
I  will get all the little changes finished and then start work on the "patch" if the patch is done before you are ready to merge i'll do it as a patch later otherwise would be nice to get in here.